rust-near-defuse-indexer/src/event_handler.rs

```diff
@@ -25,14 +26,33 @@ async fn handle_streamer_message(message: StreamerMessage, client: &Client) {
         .shards
         .iter()
         .flat_map(|shard| {
-            shard.receipt_execution_outcomes.iter().flat_map(|outcome| {
+            let chunk = Option::as_ref(&shard.chunk).unwrap();
+            let header = message.block.header.clone();
+            
+            shard.receipt_execution_outcomes.iter().flat_map(move |outcome| {
```

Comment:
I think this might work partially, only for receipts that were originated by a transaction in the same block. 

Please check this implementation where a cache is used to capture tx from previous blocks: https://github.com/near/near-indexer-for-explorer/blob/master/database/src/adapters/receipts.rs#L159